bee-ledger/src/white_flag.rs

```diff
@@ -236,17 +236,22 @@ where
 
         match tangle.get(message_id).await {
             Some(message) => {
-                let parent1 = message.parent1();
-                let parent2 = message.parent2();
+                let mut next = None;
 
-                if visited.contains(parent1) && visited.contains(parent2) {
-                    on_message(storage, message_id, &message, metadata).await?;
-                    visited.insert(*message_id);
-                    messages_ids.pop();
-                } else if !visited.contains(parent1) {
-                    messages_ids.push(*parent1);
-                } else if !visited.contains(parent2) {
-                    messages_ids.push(*parent2);
+                for parent in message.parents() {
+                    if !visited.contains(parent) {
+                        next.replace(parent);
+                        break;
+                    }
+                }
+
+                match next {
+                    Some(next) => messages_ids.push(*next),
+                    None => {
+                        on_message(storage, message_id, &message, metadata).await?;
+                        visited.insert(*message_id);
+                        messages_ids.pop();
+                    }
```

Comment:
this is more of a comment. The code will work **as long** as no one messes with the message builder, or allows to bypass it. Out of interest: Is it possible that a pen tester could create a specially crafted messages that gets deserialized into a Message with an empty parents vec? Because for such messages the `None` case is executed also.

Reply:
It's not possible no. You have to announce the number of parents in a message, if you put 0, it is directly rejected.